contents/functionalProgramming/intro.md

```diff
@@ -50,32 +50,32 @@ We see that the impure function violates both conditions. Running `addXToY` mult
 
 #### Immutability
 
-Another concept in Functional programming is immutability. When we assign `x=1`, it does not make mathematical sense to have this expression `x:=x+1`.
+Another concept in functional programming is immutability. When we assign `x=1`, it does not make mathematical sense to have this expression `x:=x+1`.
 
-Having immutability built into Functional programming languages like Haskell also helps to ensure that functions are `pure` because mutable variables and states can introduce side-effects.
+Having immutability built into functional programming languages like Haskell also helps to ensure that functions are `pure` because mutable variables and states can introduce side-effects.
 
 To know more about immutability in functional languages, you can take a look at:
 
 - [Immutability in Haskell](https://mmhaskell.com/blog/2017/1/9/immutability-is-awesome)
 - [Immutability in Elm](http://elmprogramming.com/immutability.html)
-- [Why does immutability enable Functional programming](https://stackoverflow.com/questions/12207757/why-do-immutable-objects-enable-functional-programming)
+- [Why does immutability enable functional programming](https://stackoverflow.com/questions/12207757/why-do-immutable-objects-enable-functional-programming)
 
 ## Techniques in Functional Programming
 
 #### Recursion
 
-Pure functional languages does not have loop constructs that procedural languages does. This is because `Iteration` usually involves state mutation per iteration. Since Functional programming avoids state changes, `Recursion` is a commonly used [technique](https://www.quora.com/Why-dont-pure-functional-programming-languages-provide-a-loop-construct) to replace `Iteration`.
+Pure functional languages does not have loop constructs that procedural languages does. This is because `Iteration` usually involves state mutation per iteration. Since functional programming avoids state changes, `Recursion` is a commonly used [technique](https://www.quora.com/Why-dont-pure-functional-programming-languages-provide-a-loop-construct) to replace `Iteration`.
```

Comment:
Suggestion: Pure functional languages do not have loop constructs like imperative languages.